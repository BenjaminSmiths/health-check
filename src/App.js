import React, {Component} from "react";
import logo from "./logo.svg";
import Label from "./components/label";
import "./App.css";
import Checkbox from "./components/checkbox";

class App extends Component {

    constructor(props) {
        super(props);

        this.state = {
            label: 'check the box to win'
        }
    }

    onChecked (checked) {
        let label = checked ? 'you are a winner' : 'but now you lost it :p';
        this.setState({label: label, checked: checked});
    }

    render() {
        return (
            <div className="App">
                <div className="App-header">
                    <img src={logo} className="App-logo" alt="logo"/>
                    <h2>Welcome to The Health Check</h2>

                </div>
                <div className="content-body">
                    <Label value={this.state.label}>
                        <Checkbox name="FirstOfHisName" checked={this.state.checked} onChange={(name, checked) => {
                            this.onChecked(checked);
                        }}/>
                    </Label>
                </div>
            </div>
        );
    }
}


export default App;
