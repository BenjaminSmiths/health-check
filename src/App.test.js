import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { mount } from 'enzyme';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('App has a default label', () => {
    // GIVEN
    const checkbox = mount(
        <App />
    );

    // WHEN
    checkbox.update();

    // THEN
    expect(checkbox.state().label).toEqual('check the box to win');
});

it('app has a label and checkbox', () => {
    const tree = renderer.create(
        <App />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});