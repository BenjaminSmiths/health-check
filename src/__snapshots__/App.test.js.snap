// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app has a label and checkbox 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to The Health Check
    </h2>
  </div>
  <div
    className="content-body"
  >
    <div
      style={
        Object {
          "alignContent": "space-between",
          "alignItems": "center",
          "color": "white",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <h3>
        check the box to win
      </h3>
      <div
        onClick={[Function]}
        style={
          Object {
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 20,
            "margin": 5,
            "width": 20,
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
